### PR TITLE
refactor: replace "if else" by "switch" && compact code "if else"

### PR DESCRIPTION
**Pull request by CoderI421**
#### refactor: replace "if else" by "switch" && compact code "if else"
 - replace "if else" by "switch" in the file cluster/cluster.go
 - compact code "if else" in the file redis/client/client.go

Limited ability, make it better~

### CODE REVIEW

File: redis/client/client.go [hunk 1]
```diff
@@ -185,10 +184,9 @@ func (client *Client) doRequest(req *request) {
 	}
 	if err == nil {
 		client.waitingReqs <- req
```
Comment (HDT3213) [inline redis/client/client.go:186]: missing return

**Comment by HDT3213** [comment]
This change failed to pass CI, please check: https://app.travis-ci.com/github/HDT3213/godis/builds/258175814
It caused by a code error in redis/client/client.go.

Actually, I think replacing if-else with switch makes no difference. So, I prefer to reject this pr.

**Comment by CoderI421** [comment]
> 
Ok~ Never mind, you can reject this pr ~

I just saw the operation of replacing “else if” with "switch case" in your commit("[refactor redis protocol parser](https://github.com/HDT3213/godis/commit/72c263969c4e967a3a81cc67e68022fdf6e39951)" ) , so I thought you wanted like that, so there was this submission.

Have a nice day~

**Comment by HDT3213** [comment]
The  actual author of that commit is a friend from cnblogs.com who's fork has too much difference to open a pr. So I committed that change in his place.

The major difference of that commit is to remove read state and to parse a reply in one loop. I accept both "if else" and "switch case",  yet I don't think they are enough to stand alone as a commit.

**Comment by CoderI421** [comment]
> The actual author of that commit is a friend from cnblogs.com who's fork has too much difference to open a pr. So I committed that change in his place.
> 
> The major difference of that commit is to remove read state and to parse a reply in one loop. I accept both "if else" and "switch case", yet I don't think they are enough to stand alone as a commit.

Got it~

**Pull request closed without merging**